Exception - too many connections

I had this error during the tests (Malika and Alix were testing in parallel) : 

SQL State : 53300
2017-11-21T15:14:34.083+01:00Error Code : 0
2017-11-21T15:14:34.083+01:00Message : FATAL: too many connections for role "uvrckdzq0pucckg3l02r"
2017-11-21T15:14:34.083+01:00at org.flywaydb.core.internal.util.jdbc.JdbcUtils.openConnection(JdbcUtils.java:57) ~[flyway-core-4.2.0.jar!/:na]
2017-11-21T15:14:34.083+01:00at org.flywaydb.core.Flyway.execute(Flyway.java:1418) ~[flyway-core-4.2.0.jar!/:na]
2017-11-21T15:14:34.083+01:00at org.flywaydb.core.Flyway.migrate(Flyway.java:971) ~[flyway-core-4.2.0.jar!/:na]
2017-11-21T15:14:34.084+01:00at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:216) ~[postgresql-9.4.1212.jar!/:9.4.1212]
2017-11-21T15:14:34.084+01:00at org.postgresql.Driver.makeConnection(Driver.java:404) ~[postgresql-9.4.1212.jar!/:9.4.1212]
2017-11-21T15:14:34.084+01:00at org.postgresql.Driver.connect(Driver.java:272) ~[postgresql-9.4.1212.jar!/:9.4.1212]
2017-11-21T15:14:34.084+01:00Caused by: org.postgresql.util.PSQLException: FATAL: too many connections for role "uvrckdzq0pucckg3l02r"
2017-11-21T15:14:34.084+01:00at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2455) ~[postgresql-9.4.1212.jar!/:9.4.1212]
2017-11-21T15:14:34.084+01:00at org.postgresql.core.v3.QueryExecutorImpl.readStartupMessages(QueryExecutorImpl.java:2586) ~[postgresql-9.4.1212.jar!/:9.4.1212]
2017-11-21T15:14:34.084+01:00at org.postgresql.core.v3.QueryExecutorImpl.<init>(QueryExecutorImpl.java:113) ~[postgresql-9.4.1212.jar!/:9.4.1212]
2017-11-21T15:14:34.084+01:00at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:222) ~[postgresql-9.4.1212.jar!/:9.4.1212]
2017-11-21T15:14:34.084+01:00at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:112) ~[HikariCP-2.6.3.jar!/:na]
2017-11-21T15:14:34.084+01:00at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:52) ~[postgresql-9.4.1212.jar!/:9.4.1212]
2017-11-21T15:14:34.085+01:00at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:118) ~[HikariCP-2.6.3.jar!/:na]
2017-11-21T15:14:34.085+01:00at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:358) ~[HikariCP-2.6.3.jar!/:na]
2017-11-21T15:14:34.085+01:00at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:201) ~[HikariCP-2.6.3.jar!/:na]
2017-11-21T15:14:34.085+01:00at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:443) ~[HikariCP-2.6.3.jar!/:na]
2017-11-21T15:14:34.085+01:00at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:514) ~[HikariCP-2.6.3.jar!/:na]
2017-11-21T15:14:34.085+01:00at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:111) ~[HikariCP-2.6.3.jar!/:na]
2017-11-21T15:14:34.085+01:00at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:97) ~[HikariCP-2.6.3.jar!/:na]
2017-11-21T15:14:34.086+01:00at org.flywaydb.core.internal.util.jdbc.JdbcUtils.openConnection(JdbcUtils.java:51) ~[flyway-core-4.2.0.jar!/:na]
2017-11-21T15:14:34.086+01:00... 31 common frames omitted
2017-11-21T15:14:34.788+01:00Deploy failed

--- a/README.md
+++ b/README.md
@@ -13,6 +13,9 @@ You need to [install PostgreSQL](https://www.postgresql.org/download/) on your m
 When this is done, run the database creation script:
 
     psql -h localhost -U postgres -f backend/database/database.sql
+
+Clever Cloud produces backups twice a day. You can import a backend locally to test with the real production data
+by executing the script `backend/scripts/importBackupLocally.sh`. (Read the instructions inside the script).
     
 ### Backend
 

--- a/backend/scripts/importBackupLocally.sh
+++ b/backend/scripts/importBackupLocally.sh
@@ -1,0 +1,13 @@
+#!/bin/sh
+
+# This script allows importing a backup produced by Clever Cloud in the local development database.
+
+# The first step is to download a backup locally, from the Clever Cloud console - PostgreSQL addon
+# (https://console.clever-cloud.com/organisations/orga_dd753560-9dfe-4c93-a891-c639d138354b/addons/addon_9fbf3f9f-4f62-4c41-a086-37990d898f33)
+#
+# You can then clean your local database schema using ./gradlew/flywayCleanInteg, and then import the downloaded backup
+# by executing this script with the downloaded dump file path as argument.
+#
+# The script will ask for the local database password: globe42
+
+pg_restore -h localhost -p 5432 -U globe42 -d globe42 --format=c --no-owner $1

--- a/backend/src/main/resources/application.yml
+++ b/backend/src/main/resources/application.yml
@@ -15,7 +15,7 @@ spring:
     username: globe42
     password: globe42
     hikari:
-      maximum-pool-size: 5
+      maximum-pool-size: 4
   servlet:
     multipart:
       max-file-size: 10MB

